src/rpc/rawtransaction.cpp

```diff
@@ -3104,31 +3126,24 @@ UniValue rawreissueasset(const JSONRPCRequest& request)
             throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Invalid asset address provided: %s", asset_address_uni.get_str()));
         }
 
-        int input_index = -1;
+        size_t input_index = -1;
```

Comment:
what did you mean by this

Reply:
... that is a good question. Let me look at that.

Reply:
I believe the answer is this:

- The initialization of `input_index` was irrelevant before, and is still irrelevant, because all codepaths either set it or throw an exception, so the initialization value is never used anywhere;
- I believe that what I did here (unintentionally) does have defined semantics, and will result in the value of `i` being the maximum representable `size_t` (irrelevant due to the above, except for the fact that it is not undefined behavior.)
- Making this unsigned also renders the `if (input_index < 0) {` below into dead code; I'm surprised there wasn't a warning about that.
- But it also means that negative values returned from `input_index_o.get_int();` should trip the following check `else if (input_index >= mtx.vin.size())`, and so should still fail.

So I think the resulting code is correct but slightly stupid.

What I was _thinking_ here is that I have a general preference for using `size_t` rather than `int` for array indices, but obviously that is stupid here if `input_index_o.get_int()` returns `int` and can return a negative value on failure.

So probably `input_index` should be changed back to `int` (or to `ssize_t` if we believe in those.)

Reply:
There is a warning about the `< 0` in current gcc, FYI, which is how I noticed this.

Ok, cool, we can change it back to int in a followup PR. I'll leave the warning in place for the remainder of the rebase since we're almost done anyway.